src/App.js

```diff
@@ -1,23 +1,72 @@
-import logo from './logo.svg';
+import { useState } from 'react';
+import contactArr from './contacts.json'
 import './App.css';
 
+
+
 function App() {
+  let fiveContacts = contactArr.slice(0, 5)
+
+  const [contacts, setContacts] = useState(fiveContacts);
+
+  let otherContacts = contactArr.slice(5, contactArr.length)
+  let addNewContact = () => {
+
+
+    let otherContacts = contactArr.filter((newContact) => {
+      return !contacts.includes(newContact)
+    })
+
+    let randomIndex = otherContacts[Math.floor(Math.random() * otherContacts.length)]
+    setContacts(contacts.concat(randomIndex))
+  }
+
+  let sortPopularity = () => {
+   contacts.sort((a, b) => (a.popularity < b.popularity) ? 1 : -1)
+    setContacts(sortPopularity)
+  }
+
```

Comment:
Here you cannot call your function as argument in your function. 
Better make copy of your contacts before sort, and then setContacts with this new sortedContacts:

```
const sortByPopularity = () => {
  const sortedContacts = [...contacts.sort((a,b) => (a.popularity < b.popularity) ? 1 : -1)];

  setContacts(sortedContacts);
};
```